### PR TITLE
Add extension to create sitemaps

### PR DESCRIPTION
**Pull request by djwfyi**
This PR adds the `sphinx_sitemap` to the list of extensions in default-conf.py.
The extension creates a sitemap.xml file at the root of the generated HTML output at the end of an HTML build.

This allows us to create the sitemap.xml file for each product/platform.
We'll need the website team to add a sitemap-index.xml file at the root of min.io that links to each of the product/platform doc sitemaps.

### CODE REVIEW

File: sitemap_index.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+   <sitemap>
+      <loc>https://www.min.io/docs/linux/sitemap.xml</loc>
```
Comment (ravindk89) [inline sitemap_index.xml:5]: We might as well update these to what we have in #minio-docs pending AB's final sign-off

File: source/default-conf.py [hunk 2]
```diff
@@ -101,8 +101,11 @@
 exclude_patterns = ['includes/*', '*-template.rst']
 
 # template for adding custom exclude paths if necessary for a given tag
+# html_baseurl is used by sphinx_sitemap extension to generate a sitemap.xml for each platform.
+# The sitemaps are combined in a sitemapindex.xml file at the root level.
 
 if tags.has("linux"):
+    html_baseurl = 'https://www.min.io/docs/linux/'
```
Comment (ravindk89) [inline source/default-conf.py:108]: same w.r.t. the expected base URLs.

**Review by ravindk89 — APPROVED** [review]
LGTM mod nits

**Review by ravindk89 — APPROVED** [review]
LGTM - one thing to double check is that we need the trailing slash on `html_baseurl`. If the sitemap URLs all look fine then you are good to go

File: source/default-conf.py (hunk 2 above)
Comment (djwfyi) [reply source/default-conf.py]: Fixed

File: sitemap_index.xml (hunk 1 above)
Comment (djwfyi) [reply sitemap_index.xml]: Updated

**Comment by djwfyi** [comment]
The example on https://github.com/jdillard/sphinx-sitemap has a trailing slash, so I'm leaving it.

**Comment by ravindk89** [comment]
:shipit:

**Pull request merged**